### PR TITLE
Enable ruff `flake8-builtins` rule

### PR DESCRIPTION
**Pull request by CoolCat467**
This pull request enables ruff's `flake8-builtins` rule, sorts the enabled rules and ignored rules sections, and fixes or ignores a lot of newly generated "argument or variable is shadowing a python builtin" errors.

### CODE REVIEW

**Comment by Fuyukai** [comment]
Why is this three separate PRs that will all conflict due to all three reformattting the pyproject.toml?

**Comment by CoolCat467** [comment]
Because they are separate rules, and if they conflict I will deal with that.

**Comment by CoolCat467** [comment]
The segfault in pypy-3.9 happened again: https://github.com/python-trio/trio/actions/runs/7607032357/job/20713702030#step:5:1186
For future reference, where should these be recorded?

**Comment by A5rocks** [comment]
Gah, well see the segfaults don't exactly provide much information... But I agree that it's annoying. At this point ig we might as well make an issue here (on trio) to track things cause upstream pytest hasn't been able to solve this yet.

---

Also, I don't really see the use of this rule. We have type checkers; I don't see the issue with this shadowing. Maybe others see issue though?

**Comment by CoolCat467** [comment]
In all reality we might not need `flake8-builtins`, but I wanted to throw it out there because it's a part of best practice not to shadow builtins.

**Comment by jakkdl** [comment]
I'm in favor of not shadowing builtins in general, type checking might catch most errors it might lead to (only if the object is different though!) - but I can definitely see it causing confusion/lead to bugs/etc.

File: pyproject.toml [hunk 1]
```diff
@@ -128,6 +129,8 @@ extend-exclude = [
 'src/trio/lowlevel.py' = ['F401']
 'src/trio/socket.py' = ['F401']
 'src/trio/testing/__init__.py' = ['F401']
+'src/trio/_core/_generated_io_kqueue.py' = ['A002']
+'src/trio/_core/_io_kqueue.py' = ['A002']
```
Comment (jakkdl) [inline pyproject.toml:133]: probably good to have comments here (for the F401 too) saying what the rule is and why it's ignored.

File: src/trio/_socket.py [hunk 2]
```diff
@@ -356,14 +356,16 @@ def socket(
...
+            family, type, proto, fileno
+        )
     stdlib_socket = _stdlib_socket.socket(family, type, proto, fileno)
     return from_stdlib_socket(stdlib_socket)
 
 
 def _sniff_sockopts_for_fileno(
     family: AddressFamily | int,
-    type: SocketKind | int,
+    type_: SocketKind | int,  # name shadowing builtin
```
Comment (jakkdl) [inline src/trio/_socket.py:368]: Don't think there's a need for a long-lasting comment here.

File: src/trio/_core/_io_kqueue.py [hunk 3]
```diff
@@ -118,7 +118,9 @@ def current_kqueue(self) -> select.kqueue:
     @contextmanager
     @_public
     def monitor_kevent(
-        self, ident: int, filter: int
+        self,
+        ident: int,
+        filter: int,  # Name shadows builtin
```
Comment (jakkdl) [inline src/trio/_core/_io_kqueue.py:123]: this comment gives zero info unless you know that it refers to A002, and that it's ignored in this file. So probably remove it, and/or add a longer comment saying that A002 is ignored in the file and why. But not sure either is warranted

**Review by jakkdl — CHANGES_REQUESTED** [review]
actually, after reading through all the diffs I think we should ignore `A002` - but we could keep `A001`. We're never going to change the argument names for external-facing functions, and I'm not sure it's worth the need for lots of ugly `noqa`s to be extra safe on internal functions.

File: src/trio/_core/_io_kqueue.py (hunk 3 above)
Comment (CoolCat467) [reply src/trio/_core/_io_kqueue.py]: Changed in 67116bc

File: src/trio/_socket.py (hunk 2 above)
Comment (CoolCat467) [reply src/trio/_socket.py]: Changed in 67116bc

File: pyproject.toml (hunk 1 above)
Comment (CoolCat467) [reply pyproject.toml]: Changed in 67116bc

**Review by jakkdl — APPROVED** [review]
Looks great!

**Review by TeamSpen210 — APPROVED** [review]

**Pull request merged**